### PR TITLE
New martial arts style- Drunk Brawling

### PR DESCRIPTION
**Pull request by tigercat2000**
This adds a new martial arts datum, labeled as Drunken Brawling. This
style is automatically added and removed to people via ethanol,
specifically, 10 life ticks after they start slurring their words.

The style is mostly debuffs, as there is a flat 50% chance to miss any
harm-intent hit, a 70% chance to fail to grab, and disarm is unaffected.

The actual 'features' of this style can be summarized as such:
 - Custom attack verbs
   - "jab","uppercut","overhand punch",
   "drunken right hook","drunken left hook"
   - Uppercut has a 5% chance to do 14 damage

 - Limits combat abilities
   - 70% chance to fail to passively grab
   - 50% chance to miss any harm intent
   - Disarm intent is untouched.

 - Added by ethanol when someone has had it into their system
   for a while
   - Removed as soon as the ethanol becomes lower than 3 units

Player notes:
 - When you get (very) drunk, you will now obtain this style, which can be either very bad, or very good.
 - Bar fight, anyone? Come on, Betcha can't even hit me!

Admin notes: 
 - Automatically obtained from drinking ethanol and subtypes (anything at the bar)

Coder notes: 
 - An actual (common) use for the martial arts system
 - Always overrides harm intent, intentionally, to prevent people from just falling back to attackhand if they get caught in the failure probability
 - Uppercut is extremely rare, but does over twice the damage as a normal punch

### CODE REVIEW

**Comment by Nienhaus** [comment]
10/10

**Comment by Iamgoofball** [comment]
:+1:

File: code/datums/martial.dm [hunk 1]
```diff
@@ -120,6 +120,66 @@
...
+
+	D.apply_damage(damage, BRUTE, null, armor_block)
+	D.apply_effect(damage, STAMINA, armor_block)
+	if(D.getStaminaLoss() > 50)
+		var/knockout_prob = D.getStaminaLoss() + rand(-15,15)
+		if((D.stat != DEAD) && prob(knockout_prob))
+			D.visible_message("<span class='danger'>[A] has knocked [D] out with a haymaker!</span>", \
+								"<span class='userdanger'>[A] has knocked [D] out with a haymaker!</span>")
+			D.apply_effect(10,WEAKEN,armor_block)
+			D.SetSleeping(5)
```
Comment (Fox-McCloud) [inline code/datums/martial.dm]: use paralyse and not sleeping.

Comment (tigercat2000) [inline code/datums/martial.dm]: sleeping is actually consistent with the boxing style, but alright

**Pull request merged**